File: 2021W29/B0neh3ad/pgr_42883.py
def solution(number, k):
    s, e = 0, k+1
    answer = ''
    for h in range(len(number)-k-1):
        mchar = '0'
        midx = 0
        print("s:", s, ", e:", e)
        for i in range(s,e):
            if mchar < number[i]:
                mchar = number[i]
                midx = i
        answer += mchar
        s = midx + 1
        e = s + (k+h-midx+1) if (k+h-midx > 0) else (len(number))

    return answer

print(solution("1231234", 2))

File: 2021W29/B0neh3ad/pgr_42883.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(const pair<int, int> &lhs, const pair<int, int> &rhs){
    return (lhs.first != rhs.first) ? (lhs.first > rhs.first) : (lhs.second < rhs.second);
}

string solution(string number, int k) {
    string answer = "";
    vector<pair<char, int> > num(number.size());
    
    for(int i=0; i<number.size(); ++i)
        num[i] = make_pair(number[i], i);

    sort(num.begin(), num.end(), cmp);

    int stand = num.size() - k;
    for(int i=0; i<num.size(); ++i){
        cout << num[i].second << " vs " << stand << " ";
        if(num[i].second >= stand){
            answer += num[i].first;
            stand = num[i].second;
        }
        cout << answer << endl;
    }

    return answer;
}

int main(void){
    cout << solution("1924", 2);
    return 0;
}